### PR TITLE
Fix UI elements related to OAuth / Bearer token validation

### LINKED ISSUE #21325: Loading spinner for OAuth / Bearer token validation broken
### Explain what happens

1. Configure Cockpit to use OAuth and Bearer tokens to log in
2. Open Cockpit, get redirected to the OAuth provider and back to Cockpit
3. While waiting for token validation, observe this odd icon which is probably supposed to be a loading spinner:

Screenshot 2024-11-25 at 9 30 52 PM

### Version of Cockpit

329

### Where is the problem in Cockpit?

None

### Server operating system

Fedora

### Server operating system version

Fedora CoreOS 41.20241027.3.0

### What browsers are you using?

Chrome, Safari macOS, Safari on iPhone

### System log

_No response_

### PR DESCRIPTION
**Pull request by marvinruder**
* Fixes #21325

### CODE REVIEW

**Comment by marvinruder** [comment]
It looks like this loading spinner was never supposed to work outside of a certain button. The animation is fixed when applying `display: inline-flex` to every spinner, not just to those inside a `.pf-v5-c-button[spinning]`:

Screenshot 2024-11-25 at 9 46 23 PM

File: pkg/static/login.scss [hunk 1]
```diff
@@ -251,7 +251,7 @@ label.checkbox {
   display: inline-block;
 }
 
-.pf-v5-c-button[spinning] .spinner {
+.spinner {
```
Comment (martinpitt) [inline pkg/static/login.scss:254]: Thanks @marvinruder for fixing this! However, I think this breaks the other "spinner" in the "login-actions" form. That one is enabled/disabled dynamically:

pkg/static/login.js:        id("login-button").removeAttribute('spinning');
pkg/static/login.js:        id("login-button").setAttribute('spinning', "true");

To make this more targetted, I suppose this should keep the original selector, but add the new one:

```css
.pf-v5-c-button[spinning] .spinner,
#login-wait-validating .spinner {
```

**Review by martinpitt — CHANGES_REQUESTED** [review]

**Comment by marvinruder** [comment]
As mentioned in #21325, I adressed some more UI element issues:

- The selector `.login-pf .container .help-block` appears to apply only to the “Validating authentication token” text. The `help-block` class was probably applied to easily get `display: block` and a pretty margin. To fix the color, `#fff` was specified, which works in dark mode, but not in light mode. `color: unset` will reset the color to the default values, which work nicely on the background of both themes
- The spinner color is always white, which works in buttons on both themes, but not on the plain white background of the bright theme. `html:not(.pf-v5-theme-dark) #login-wait-validating .spinner` selects a spinner only when it is the one in the “Validating authentication token” dialog and only if the dark theme is not active.
- `.pf-v5-theme-dark a.pf-v5-c-button` selects links that are styled as buttons with the dark theme active, and applies the default text color `#fff`, which works great with the button styling. Without it, the button link text has a light blue color in dark mode, which looks faded.

**Review by martinpitt — APPROVED** [review]
Thanks a lot! This looks reasonable and doesn't break the non-oauth login page.

**Pull request merged**